x/ibc/light-clients/solomachine/types/proof.go

```diff
@@ -23,193 +22,292 @@ func VerifySignature(pubKey crypto.PubKey, data, signature []byte) error {
 	return nil
 }
 
-// EvidenceSignBytes returns the sign bytes for verification of misbehaviour.
-//
-// Format: {sequence}{data}
-func EvidenceSignBytes(sequence uint64, data []byte) []byte {
-	return append(
-		sdk.Uint64ToBigEndian(sequence),
-		data...,
-	)
+// MisbehaviourSignBytes returns the sign bytes for verification of misbehaviour.
+func MisbehaviourSignBytes(cdc codec.BinaryMarshaler, sequence uint64, data []byte) ([]byte, error) {
+	signBytes := &SignBytes{
```

Comment:
should I diversifier and timestamp?

Reply:
yes, misbehaviour signBytes must match how the other signBytes are signed otherwise we will fail to verify misbehaviour. Does that make sense? That is why I initially proposed the wrapper, so they had a common interface to construct the signBytes while still allowing some variance in `Data`